lib/app.js

```diff
@@ -160,29 +273,142 @@ class AppView extends React.Component {
     );
   }
 
-  platform(value) {
-    return Platform.OS === value
-  }
-
   onNetworkAlertClosed() {
     setTimeout(() => this.networkAlertOpen = false, 500)
   }
 
+  /**
+   * Called when the user refreshes the list
+   * via a pull-to-refresh gesture.
+   */
   onRefresh() {
     debug('on refresh');
     track.pullRefresh();
     this.stopScanning();
-    actions.clearItems();
+    this.props.clearItems();
     this.startScanning();
   }
+
+  /**
+   * Called when the navigator finishes
+   * a transition to a new 'scene'.
+   */
+  onNavigated({ type }) {
+    debug('navigated', type);
+    track.view(type);
+  }
+
+  /**
+   * Called when a list-item is pressed.
+   *
+   * Depending on what prefs are set, this
+   * could open the expanded view, or simply
+   * navigate to the item's url.
+   */
+  onItemPress(item) {
```

Comment:
An item press can result in an app navigation (when expanded view is preffed on), so we do have to bubble this event up the the app level sometimes. Perhaps though we could handle logic at the list view level and only bubble up the app view level when need be.

That said (I'm thinking out loud :)) we could use the passed `navigator` object into the `ListScene` and let it do the navigation. Or we also may be able to do routing via a rdux state change.

So many options!!